src/lib/translations.js

```diff
@@ -0,0 +1,88 @@
+import i18n from "sveltekit-i18n";
+import languageIndex from "./lang/index.json";
+
+/** @type {import("sveltekit-i18n").Config} */
+const config = ({
+    loaders: [
+        {
+            locale: "en",
+            key: "common",
+            loader: async () => (
+                await import("./lang/en/common.json")
+            ).default,
+        },
+        {
+            locale: "en",
+            key: "home",
+            routes: ["/"],
+            loader: async () => (
+                await import("./lang/en/home.json")
+            ).default,
+        },
+        {
+            locale: "ja",
+            key: "common",
+            loader: async () => (
+                await import("./lang/ja/common.json")
+            ).default,
+        },
+        {
+            locale: "ja",
+            key: "home",
+            routes: ["/"],
+            loader: async () => (
+                await import("./lang/ja/home.json")
+            ).default,
+        },
+        {
+            locale: "ko",
+            key: "common",
+            loader: async () => (
+                await import("./lang/ko/common.json")
+            ).default,
+        },
+        {
+            locale: "ko",
+            key: "home",
+            routes: ["/"],
+            loader: async () => (
+                await import("./lang/ko/home.json")
+            ).default,
+        }
+    ],
+
+/*
+Why do I have to do this? I can't even do this automatically because of TS
+type checking. (can't upcast to `any` because this file is JS)
+```js
+    const syncTranslations = {};
+    Object.keys(languageIndex).forEach(key => {
+        syncTranslations[key] = languageIndex
+    });
+```
+Every language should be listed in their own language,
+so the native speakers can recognize it without being able to read the
+currently selected language.
+
+This is the only real good use for synchronous translations I can think of,
+so why does it have to be like this?
+It's even used like this in the official examples. Why?
+https://github.com/sveltekit-i18n/lib/blob/b09444c66df7cf1d34852b70893b9f93d82d4cb9/examples/one-page/src/lib/translations/index.js#LL6C1-L9C5
+^ In case you can't see what lines this is referencing because you don't
+have the GitHub beta enabled, this is referencing lines 6-9.
+
+If I'm doing this wrong, PLEASE tell me.
+- @CenTdemeern1
+*/
+    translations: {
+        en: languageIndex,
+        ja: languageIndex,
+        ko: languageIndex,
+        nl: languageIndex,
+        de: languageIndex,
+        id: languageIndex,
+        ru: languageIndex
+    }
+});
```

Comment:
To update the list of translations automatically you need to convert your `translations.js` file to TypeScript, then do something like this:

```ts
interface translationsObject {
    [key: string]: any,
}

const syncTranslations: translationsObject = {}

Object.keys(languageIndex).forEach(key => {
    syncTranslations[key as keyof translationsObject] = { languageIndex };
});
```

Another problem here is that you're assigning the lang object directly to a key like this `en: languageIndex`, you want to do this instead `en: { languageIndex }` which will results in an object like this:
```ts
en: {
    'languageIndex.en': 'English',
    'languageIndex.ja': '日本語',
    'languageIndex.ko': '한국어',
    'languageIndex.nl': 'Nederlands',
    'languageIndex.de': 'Deutsch',
    'languageIndex.id': 'Bahasa Indonesia',
    'languageIndex.ru': 'Русский язык',
},
```
instead of
```ts
en: {
    en: 'English',
    ja: '日本語',
    ko: '한국어',
    nl: 'Nederlands',
    de: 'Deutsch',
    id: 'Bahasa Indonesia',
    ru: 'Русский язык',
}
```

svelte-i18n TypeScript example is here https://github.com/sveltekit-i18n/lib/blob/master/examples/parser-icu/src/lib/translations/index.ts

Reply:
I tried using a typescript file but the library threw all kinds of errors when I did. I'll try again.

Reply:
I also had some errors but they disappeared after I started dev server, it's weird

Reply:
Also here's a better example that uses the same parser as it does now: https://github.com/sveltekit-i18n/lib/blob/master/examples/parser-default/src/lib/translations/index.ts

Reply:
Is this alright? (9047a3b) I used `{lang: typeof lang}` instead of `any` because we're disallowing explicit any

Reply:
LGTM :)